app/src/main/java/org/oppia/android/app/topic/TopicFragmentPresenter.kt

```diff
@@ -105,10 +106,12 @@ class TopicFragmentPresenter @Inject constructor(
         }
       }
     }.attach()
-    if (topicId.isNotEmpty() && storyId.isNotEmpty())
-      setCurrentTab(TopicTab.LESSONS)
-    else if (topicId.isNotEmpty() && storyId.isEmpty())
-      setCurrentTab(TopicTab.INFO)
+    if (!isConfigChanged) {
+      if (topicId.isNotEmpty() && storyId.isNotEmpty())
+        setCurrentTab(TopicTab.LESSONS)
+      else if (topicId.isNotEmpty() && storyId.isEmpty())
+        setCurrentTab(TopicTab.INFO)
+    }
```

Comment:
I think we can directly use a variable like `previousActiveTab` and use the TopicTab in this right. 
`isConfigChanged` doesn't look good to me.

Reply:
@FareesHussain can you please explain your approach a bit more.

Reply:
Instead of saving the boolean save the current `TopicTab` and use it in the onSavedInstance(using `addProto`)

Reply:
I think viewpager retains the tab on config change. The problem here was caused because we were changing the tab to the lessons tab every time the activity is created. So I think saving the tab is not be necessary. WDYT

Reply:
@MaskedCarrot Your explanation is correct in my opinion.

Reply:
@BenHenning I discussed about the first comment that @FareesHussain has mentioned. He informed me that you have suggested to use `savedInstance` to save the values instead of using `configChanged` [Reference](      app:setText="@{viewModel.answerTe}"
). Can you mention the reason why we prefer that over the solution done in this PR.

Reply:
We have to be careful when restoring state & make sure that everything we're restoring is ultimately stored in Android's activity-tied Bundle class (otherwise in low memory cases we'll lose all state-- #2571 is tracking fixing this broadly & especially for domain ephemeral state). In this case the current solution is fine since we're relying on arguments to restore state, and that should be stored in the activity-bound Bundle upon configuration change or low memory situations.

Reply:
@BenHenning Thanks.

@FareesHussain WDYT?

Reply:
LGTM